src/main/scala/ordertaking/domain/GizmoRefined.scala

```diff
@@ -0,0 +1,52 @@
+package ordertaking.domain
+
+import cats.data.ValidatedNel
+import cats.syntax.all.*
+import io.github.iltotore.iron.*
+import io.github.iltotore.iron.cats.*
+import io.github.iltotore.iron.constraint.all.*
+
+object GizmoRefined extends App {
+  // Refinement types
+  type StartsWithG = DescribedAs[StartWith["G"], "Should start with G"]
+  // Note, I had to prefix my regex with `.` to match any character, which I don't hugely like
+  type Match3Digits = DescribedAs[Match[".[0-9][0-9][0-9]"], "Should match 3 digits"]
+  // Combine refinements
+  type GizmoCode = DescribedAs[StartsWithG & Match3Digits, "Should be G followed by three digits"]
```

Comment:
Couldn't that be `DescrbiedAs["G\d{3}", "Should be G followed by three digits"]` ?